plugins/chat/spec/jobs/regular/chat/auto_join_channel_batch_spec.rb

```diff
@@ -3,251 +3,30 @@
-
-      it "only joins group members with access to the category" do
-        another_user = Fabricate(:user, last_seen_at: 15.minutes.ago)
-
-        job.execute(chat_channel_id: channel.id, starts_at: user.id, ends_at: another_user.id)
-
-        assert_users_follows_channel(channel, [user])
-        assert_user_skipped(channel, another_user)
-      end
-
-      it "works if the user has access through more than one group" do
-        second_chatters_group = Fabricate(:group)
-        Fabricate(:category_group, category: category, group: second_chatters_group)
-        second_chatters_group.add(user)
-
-        job.execute(chat_channel_id: channel.id, starts_at: user.id, ends_at: user.id)
-
-        assert_users_follows_channel(channel, [user])
-      end
-
-      it "joins every user with access to the category" do
-        another_user = Fabricate(:user, last_seen_at: 15.minutes.ago)
-        chatters_group.add(another_user)
-
-        job.execute(chat_channel_id: channel.id, starts_at: user.id, ends_at: another_user.id)
-
-        assert_users_follows_channel(channel, [user, another_user])
-      end
-
-      it "doesn't join users with read-only access to the category" do
-        restricted_category = Fabricate(:category, read_restricted: true)
-        another_user = Fabricate(:user, last_seen_at: 15.minutes.ago)
-        non_chatters_group = Fabricate(:group)
-        readonly_channel =
-          Fabricate(:category_channel, chatable: restricted_category, auto_join_users: true)
-        Fabricate(
-          :category_group,
-          category: restricted_category,
-          group: non_chatters_group,
-          permission_type: CategoryGroup.permission_types[:readonly],
-        )
-        non_chatters_group.add(another_user)
-
-        job.execute(
-          chat_channel_id: readonly_channel.id,
-          starts_at: another_user.id,
-          ends_at: another_user.id,
-        )
-
-        assert_user_skipped(readonly_channel, another_user)
-      end
-
-      it "does join users with at least one group with create_post or full permission" do
-        restricted_category = Fabricate(:category, read_restricted: true)
-        another_user = Fabricate(:user, last_seen_at: 15.minutes.ago)
-        non_chatters_group = Fabricate(:group)
-        private_channel =
-          Fabricate(:category_channel, chatable: restricted_category, auto_join_users: true)
-        Fabricate(
-          :category_group,
-          category: restricted_category,
-          group: non_chatters_group,
-          permission_type: CategoryGroup.permission_types[:readonly],
-        )
-        non_chatters_group.add(another_user)
-
-        other_group = Fabricate(:group)
-        Fabricate(
-          :category_group,
-          category: restricted_category,
-          group: other_group,
-          permission_type: CategoryGroup.permission_types[:create_post],
-        )
-        other_group.add(another_user)
+  it "logs an error on failed contract" do
```

Comment:
Usually, when referring to a condition in an `it`, it’s a hint to use a `context` :grin: 
I’d rewrite this like that instead:
```ruby
context "when contract fails" do
  before { Jobs.run_immediately! }

  it "logs an error" do
    …
  end
end
```